src/OpenTelemetry/Metrics/ExponentialHistogramBuckets.cs

```diff
@@ -58,10 +61,12 @@ public struct Enumerator
     {
         private readonly long[] buckets;
         private int index;
+        private int size;
 
-        internal Enumerator(long[] buckets)
+        internal Enumerator(long[] buckets, int size)
         {
             this.index = 0;
```

Comment:
Shouldn't we assign `index` to `Offset`?

Reply:
No, not in this case, the values from the circular buffer are always copied starting at index 0 up to the size of the circular buffer.